tools/pre-commit

```diff
@@ -1 +1,24 @@
-./tools/lint-all $(git diff --cached --name-only --diff-filter=ACM) || true
+#!/bin/sh
+
+# This hook runs the Zulip code linter ./tools/lint-all and returns true
+# regardless of linter results so that your commit may continue.
+
+# Messages from the linter will output to screen.
+#
+# If you are using the vagrant method for your Zulip dev setup, the linter will
+# be run through vagrant ssh.
+
+if `which vagrant > /dev/null` && vstatus="$(vagrant status)" && ! [[ $vstatus =~ 'not created' ]]
+then
+  vcmd='/srv/zulip/tools/lint-all $(cd /srv/zulip && git diff --cached --name-only --diff-filter=ACM) || true'
+  if ! [[ $vstatus =~ 'running' ]]
+  then
+    echo "Vagrant machine not running; lint-all skipped."
+  else
+    echo "Running lint-all using vagrant..."
+    vagrant ssh -c "$vcmd" 2> /dev/null
```

Comment:
Because for some reason, in my tests anyway, vagrant always reports to stderr that the ssh connection is closed after lint-all completes. Doesn't seem to be helpful information. I didn't investigate whether or not that's an error condition we should address.